File: bookstore-springboot-example/bookstore-service/src/main/java/example/springboot/bookstore/persistence/entity/BookEntity.java
package example.springboot.bookstore.persistence.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Data;

import java.time.LocalDate;

@Data
@Entity
@Table(name = "book")
public class BookEntity {

    @Id
    private Long id;

    private String title;

    private String author;

    private String description;

    private LocalDate releaseDate;

    private String coverImageUrl;
}

File: bookstore-service-micronaut-kotlin/src/main/kotlin/com/example/micronaut/bookstore/repository/entity/BookEntity.kt
package com.example.micronaut.bookstore.repository.entity

import jakarta.persistence.Entity
import jakarta.persistence.Id
import java.time.LocalDate

@Entity(name = "book")
class BookEntity(
    @Id val id: String,
    val title: String,
    val author: String,
    val description: String? = "",
    val releaseDate: LocalDate,
    val coverageImageUrl: String? = null
)